data-protocols/ids/ids-api-multipart-dispatcher-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/dispatcher/sender/MultipartCatalogDescriptionRequestSender.java

```diff
@@ -55,10 +55,41 @@ public MultipartCatalogDescriptionRequestSender(@NotNull String connectorId,
                                                     @NotNull ObjectMapper objectMapper,
                                                     @NotNull Monitor monitor,
                                                     @NotNull IdentityService identityService,
-                                                    @NotNull TransformerRegistry transformerRegistry) {
+                                                    @NotNull IdsTransformerRegistry transformerRegistry) {
         super(connectorId, httpClient, objectMapper, monitor, identityService, transformerRegistry);
     }
 
+    private static void createOfferResourcesFromProperties(ResourceCatalog catalog, ObjectMapper mapper) {
+        if (catalog.getProperties() != null) {
+            for (Map.Entry<String, Object> entry : catalog.getProperties().entrySet()) {
+                if ("ids:offeredResource".equals(entry.getKey())) {
+                    JsonNode node = mapper.convertValue(entry.getValue(), JsonNode.class);
+                    List<Resource> offeredResources = new LinkedList<>();
+                    for (JsonNode objNode : node.get("objectList")) {
+                        Map<String, Object> resource = new HashMap<>();
+                        resource.put("@type", "ids:Resource");
+                        resource.putAll(mapper.convertValue(objNode, Map.class));
+                        offeredResources.add(mapper.convertValue(resource, Resource.class));
+                    }
+                    catalog.setOfferedResource(offeredResources);
```

Comment:
Is it possible that more than one property maps contains key "ids:offeredResource"?

If yes, should the `List<Resource> offeredResources = new LinkedList<>();` and `catalog.setOfferedResource(offeredResources);` be outside of the main loop to save all offered resources? Otherwise they are overwritten in every iteration.

But if not and only one "ids:offeredResource" is expected - can we get the value first and then skip one loop level?

Reply:
To be honest, I don't know, the only reason I "modified" this is because of a rebase.

However, it appears to make sense to go through all the IDS classes with a fine-toothed comb at some point.

Reply:
I see, thought it's a new code

Reply:
it seems that during the rebase it was accidentally moved to the top of the class, should stop appear as a new change if will be moved back there

Reply:
This might have happened during auto formatting.

Reply:
I'll go over the files and reformat